app/controllers/pages_controller.rb

```diff
@@ -28,4 +28,16 @@ def assistant
     return not_found unless assistant_landing_page_enabled? || public_assistant_landing_page_enabled?
     render layout: "layouts/pagesv2"
   end
+
+  def help
+    file_name = params[:article_name] ? params[:article_name] : "help"
+    @file = File.read(Rails.root.join("docs", "#{file_name}.md"))
```

Comment:
We can't just blindly join with raw user input here, we're opening ourselves up to path traversal attacks. Can you find a way to sanitize this input?

```ruby
irb(main):003:0> Rails.root
=> #<Pathname:/Users/d12/dev/events_remote_loader_example>

irb(main):004:0> Rails.root.join("..")
=> #<Pathname:/Users/d12/dev>

irb(main):006:0> Rails.root.join("/etc/passwd")
=> #<Pathname:/etc/passwd>

irb(main):007:0> File.read(Rails.root.join("/etc/passwd"))
=> "##\n# User Database\n# \n# Note that this file is ..........
```

Reply:
Good catch. How does this look? https://github.com/education/classroom/pull/1794/commits/147bfe97fe2e065b4af8a4ddf41c72b49d983193